src/main/java/org/threadly/concurrent/AbstractPriorityScheduler.java

```diff
@@ -154,6 +171,47 @@ public void scheduleWithFixedDelay(Runnable task, long initialDelay, long recurr
   public void scheduleAtFixedRate(Runnable task, long initialDelay, long period) {
     scheduleAtFixedRate(task, initialDelay, period, null);
   }
+
+  protected static TaskWrapper getNextTask(QueueSet highPriorityQueueSet, QueueSet lowPriorityQueueSet, 
```

Comment:
There is the potential for this to cause more clock calls when consuming tasks in the PriorityScheduler.  Mostly because after this is called, getScheduleDelay() will be invoked on the returned task.  If the task is not know to be ready, this will cause another clock call.

The good is that this structure can avoid clock calls entirely (just our normal 100 millis update sequence), this would be the behavior if the scheduler is > 100 millis behind.  Which would be awesome.  The time when clock calls would be invoked is when the pool is NOT under full load (but as described above, we may have more clock calls than previously under those conditions).

Not a full win, but maybe a trade off that will pay out in the benchmarks.